src/hooks/useWalletConnection.ts

```diff
@@ -90,18 +90,23 @@ export const useWalletConnection = () => {
 
   const selectedNetwork = useSelector(getSelectedNetwork);
   const walletConnectConfig = ENVIRONMENT_CONFIG_MAP[selectedNetwork].wallets.walletconnect;
-
-  const { connectAsync: connectWagmi } =
-    walletType && walletConnectionType
-      ? useConnectWagmi({
-          connector: resolveWagmiConnector({
+  const wagmiConnector = useMemo(
+    () =>
+      walletType && walletConnectionType
+        ? resolveWagmiConnector({
             walletType,
             walletConnection: {
               type: walletConnectionType,
             },
             walletConnectConfig,
-          }),
-        })
+          })
+        : undefined,
+    [walletConnectConfig, walletType, walletConnectionType]
+  );
+
+  const { connectAsync: connectWagmi } =
+    walletType && walletConnectionType
+      ? useConnectWagmi({ connector: wagmiConnector })
       : useConnectWagmi();
```

Comment:
when `walletType && walletConnectionType` is not true, `wagmiConnector` is already undefined; can we just simplify this to 
```
const { connectAsync: connectWagmi } = useConnectWagmi({ connector: wagmiConnector });
```

Reply:
oh yea good call